chore(supabase): Set up Supabase as backend service and add initial tests (#83)

911samuel opened the pull request:
#### What does this PR do?

This PR sets up Supabase as the backend service for our application and includes initial tests for sign-up and sign-in functionality.

#### Description of Task to be completed?

* Initializes the Supabase client with the provided URL and Anon Key.
* Configures authentication settings for AsyncStorage, enabling auto-refresh and session persistence.
* Adds basic tests to verify sign-up and sign-in functionality.

#### How should this be manually tested?

To manually test the changes:

1. Access the sign-up page (if applicable).
2. Enter a valid email address and password.
3. Confirm the password.
4. Click the "Sign Up" button.
5. Verify that the user is redirected to the user profile page (or expected success behavior).
6. Try to sign up with an invalid email address or password.
7. Ensure appropriate error messages are displayed in these scenarios.
8. Repeat steps 1-4 for sign-in functionality.

#### Questions:

* Are there any specific requirements for error handling during sign-up and sign-in?

**Please review these changes and provide feedback.**

Latiah commented:
Good work I was able to sign up and log in

Latiah commented:
Hey @911samuel here are some few suggestions. Can you change so that when a person signs up ,he /she is directed to the sign up instead of going to the fill in your profile screen and also consider informing the user to enter their credentials(email or password)  when they try to sign up or in without filling them in.

911samuel commented:
> Hey @911samuel here are some few suggestions. Can you change so that when a person signs up ,he /she is directed to the sign up instead of going to the fill in your profile screen and also consider informing the user to enter their credentials(email or password) when they try to sign up or in without filling them in.

Thank you for your feedback, @Latiah. The issue was setting up Supabase to integrate it into our project. The sign-up and sign-in functionalities were simple demonstrations to show that Supabase is configured and ready to be used as our backend service. We will need someone to refine the sign-in and sign-up processes and also handle the forgot password functionality and other aspects of the authentication workflow

fredshema commented on app/supabase.tsx:
```diff
@@ -0,0 +1,15 @@
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { createClient } from "@supabase/supabase-js";
+
+const supabaseUrl = "https://vaxjykeihsernxqxeyiy.supabase.co";
+const supabaseAnonKey =
+  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InZheGp5a2VpaHNlcm54cXhleWl5Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3MTc0NDAzMTksImV4cCI6MjAzMzAxNjMxOX0.PbKiCmDG-U_1H0gkelN3_ze8_VE8lqIsc3pi5IH2bBA"; 
```
@911samuel can you add these in environment variables, they can be shared privately but not directly in the codebase

911samuel replied:
Thank you shema for the best advice and i worked on it you can now review again

chrysloi reviewed (approved):
Looks good!

Adadie commented:
Hey @911samuel, why do I get below screen when I run the app?
![Screenshot_20240607_090301_Expo Go](https://github.com/atlp-rwanda/commanders-rn-medica/assets/68109866/94fb0f81-d2b2-4039-9483-875d2a20b33a)

911samuel committed:
chore(supabase): Set up Supabase as backend service and add initial t…
…ests (#83)

- Initializes Supabase client with provided URL and Anon Key.
- Configures authentication settings for AsyncStorage, auto-refresh, and session persistence.
- Adds initial tests for sign-up and sign-in functionality.

911samuel commented:
> Hey @911samuel, why do I get below screen when I run the app? ![Screenshot_20240607_090301_Expo Go](https://private-user-images.githubusercontent.com/68109866/337559882-94fb0f81-d2b2-4039-9483-875d2a20b33a.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3NTk2NzQsIm5iZiI6MTcxNzc1OTM3NCwicGF0aCI6Ii82ODEwOTg2Ni8zMzc1NTk4ODItOTRmYjBmODEtZDJiMi00MDM5LTk0ODMtODc1ZDJhMjBiMzNhLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA3VDExMjI1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyOTZmNjY0ZDAwZDFhMzE2ZDBjZWM5NTU2Yzk5ZWJjZjNhM2RlYzlhYThiNWYwNTM2ODZiM2VmYWU1NzExNGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.dXPvy-VpNrGUy-ko78yHs3v7_1FGvjNlAe__SEnY2Pg)

All things are solved Please review again
